spec/controllers/prompts_controller_spec.rb

```diff
@@ -1,18 +1,126 @@
-require 'spec_helper'
+require "spec_helper"
 
 describe PromptsController do
   include LoginMacros
+  include RedirectExpectationHelper
+  let(:collection) { create(:collection) }
+  let(:open_signup) do
+    signups = create(:challenge_signup)
+    signups.collection.challenge.signup_open = true
+    signups.collection.challenge.save
+    signups
+  end
+  let(:signup) { create(:challenge_signup) }
 
   before do
     fake_login
   end
 
-  describe 'no_prompt' do
-    it 'should show an error and redirect' do
-      signups = create(:challenge_signup)
-      post :edit, collection_id: signups.collection.name
-      expect(response).to redirect_to(collection_path(signups.collection))
-      expect(flash[:error]).to eq "What prompt did you want to work on?"
+  describe "no_challenge" do
+    it "should show an error and redirect" do
+      get :show, collection_id: collection.name
+      it_redirects_to_with_error(collection_path(collection), "What challenge did you want to sign up for?")
+    end
+  end
+
+  describe "no_signup" do
+    it "should show an error and redirect" do
+      post :create, collection_id: signup.collection.name
+      it_redirects_to_with_error(collection_signups_path(signup.collection) + "/new",\
+                                 "Please submit a basic sign-up with the required fields first.")
+    end
+  end
+
+  describe "signups_closed" do
+    it "should show an error and redirect" do
+      # Login as the signup owner
+      fake_login_known_user(Pseud.find(ChallengeSignup.in_collection(signup.collection).first.pseud_id).user)
```

Comment:
What I would do here is instead of having `fake_login` in the before block up top and the `fake_login_known_user` here, is to have this in the `before` block on line 15:

```
let(:user) { create(:user) }
before do
  fake_login_known_user(user)
end
```

Then, for each context here you can just `let` user as `Pseud.find(ChallengeSignup.in_collection(signup.collection).first.pseud_id).user` or whatever it's supposed to be. It's still a line per context, but it seems like it'd adhere to the DRY principle a bit better.

Or, it would be fewer lines to set the default user in the larger context to be `Pseud.find(ChallengeSignup.in_collection(signup.collection).first.pseud_id).user` and `let` user to be `create(:user)` where it's different, but I think setting the default as `create(:user)` makes more sense.